### PR TITLE
azureconfig/v1: update operatorkit (new Resource)

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/2747

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -183,7 +183,7 @@
   revision = "4411a6307c6b844b7f9e68aaba37407b41b1c4d7"
 
 [[projects]]
-  branch = "master"
+  branch = "log-resource"
```
Comment (kopiczko) [inline Gopkg.lock]: This is the branch. I wanted to test before merging.

File: service/azureconfig/v1/resource/deployment/resource.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func New(config Config) (*Resource, error) {
 	r := &Resource{
 		certsSearcher: config.CertsSearcher,
 		cloudConfig:   config.CloudConfig,
-		logger:        config.Logger.With("resource", Name),
+		logger:        config.Logger,
```
Comment (kopiczko) [inline service/azureconfig/v1/resource/deployment/resource.go]: We don't need this now https://github.com/giantswarm/operatorkit/pull/167

File: service/azureconfig/v1/resource/deployment/resource.go [hunk 3]
```diff
@@ -226,11 +226,6 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
 	return nil
 }
 
-// Underlying returns the underlying resource.
-func (r *Resource) Underlying() framework.Resource {
-	return r
-}
-
```
Comment (kopiczko) [inline service/azureconfig/v1/resource/deployment/resource.go]: This wasn't even compiling now. Now `Resource` fulfills `framework.CRUDResourceOps`, not `framework.Resource`.

File: service/azureconfig/v1/resource_set.go [hunk 4]
```diff
@@ -100,20 +100,25 @@ func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
...
 		if err != nil {
-			return nil, microerror.Maskf(err, "resourcegroup.New")
+			return nil, microerror.Mask(err)
+		}
+
+		resourceGroupResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
 	}
```
Comment (kopiczko) [inline service/azureconfig/v1/resource_set.go]: This is the whole transition. Old resources are created as CRUDResources.

File: service/azureconfig/v1/resource_set.go [hunk 5]
```diff
@@ -180,3 +195,17 @@ func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
 
 	return resourceSet, nil
 }
+
+func toCRUDResource(logger micrologger.Logger, ops framework.CRUDResourceOps) (*framework.CRUDResource, error) {
```
Comment (kopiczko) [inline service/azureconfig/v1/resource_set.go]: Although I know it isn't everybodys favorite to do it like that it saves some lines, especially when there are many resources. This is private, so can be changed any time to a different init approach.

I'm open for suggestions.

**Comment by kopiczko** [comment]
Tested locally.

**Comment by kopiczko** [comment]
I ask for a review, I'd like to finally merge operatorkit changes and update other operators. I'd like to agree on a single one now to reduce further friction and amount of work.

File: service/azureconfig/v1/resource_set.go (hunk 5 above)
Comment (xh3b4sd) [reply service/azureconfig/v1/resource_set.go]: I would just keep it explicit.

**Pull request merged**